projects/odf_transform/odf_transform/odf_parser/odf_parser.py

```diff
@@ -165,68 +171,115 @@ def _convert_to_number(value):
+
+        # FLAG VARIABLES Detect if it is a flag column
+        is_q_flag = var.startswith("Q") and var[:1] in metadata.keys()
+        is_qqqq_flag = odf_var_name['name'] == 'QQQQ'
+        is_general_flag = odf_var_name['name'] in ['QCFF', 'FFFF']
+        is_flag_column = is_q_flag or is_qqqq_flag or is_general_flag
+
+        # Find related variable
+        if is_qqqq_flag:
+            # MLI FLAG QQQQ should apply to previous variable
+            related_variable = previous_key
+        elif is_q_flag:
+            # Q  Format is usually Q+[PCODE] of the associated variable
+            related_variable = var[1:]
+            # Make sure that related_variable do exist!
+            if related_variable not in metadata:
+                warnings.warn('{0} flag is referring to {1} which is not available as variable'
+                              .format(var, related_variable), UserWarning)
+
+        # Set previous key for the next run
+        previous_key = var
+
+        # Make sure that the flag column relates to a specific variable and try to make sure it's the right match.
+        # Try to confirm by matching either name or code
+        if is_flag_column:
+            if not is_general_flag and related_variable:
+                # Drop odf flag name prefix
+                flag_name_with_no_prefix = re.sub(r"quality\sflag.*:\s*|quality flag of ", '',
+                                                  att['original_NAME'], 1, re.IGNORECASE)
+                # Flag name do not match either variable name or code, give a warning.
+                if related_variable and \
+                        flag_name_with_no_prefix not in metadata[related_variable].get('original_NAME') and \
+                        flag_name_with_no_prefix not in metadata[related_variable].get('original_CODE'):
+                    warnings.warn(
+                        '{0}[{4}] flag was matched to referring to {1} but odf variable name[{2}] or code[{3}] do not match'
+                            .format(var,
+                                    related_variable,
+                                    metadata[related_variable].get('original_NAME'),
+                                    metadata[related_variable].get('original_CODE'),
+                                    att['original_NAME']),
+                        UserWarning)
+
+            # Standardize Flag variable attributes, related variable and add convention attributes
+            if related_variable:
+                # Add long name attribute which is generally QUALITY_FLAG: [variable it affects]
+                if 'name' in metadata[related_variable]:
+                    att['long_name'] = flag_long_name_prefix + metadata[related_variable]['name']
+                else:
+                    att['long_name'] = flag_long_name_prefix + related_variable
+
+                # Add ancillary_variables attribute
+                if 'ancillary_variables' not in metadata[related_variable]:
```

Comment:
something's missing here, look like the `else` will never get called

Reply:
elif was replace by else